From d1e759aeabab18f7dd45c240b31efeecafc6fc8e Mon Sep 17 00:00:00 2001
Message-ID: <d1e759aeabab18f7dd45c240b31efeecafc6fc8e.1757322510.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Sun, 7 Sep 2025 17:12:05 +0200
Subject: [PATCH] parse.py: adjust to clang 21

clang 21 makes `Cursor` hashable but `File` un-hashable. Adjust by using
Cursor instead of its elemnets  so that we can index the cache dictionary
again. `Cursor` has been hashable at least from clang 18.11.
---
 scripts/wrap/parse.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/wrap/parse.py b/scripts/wrap/parse.py
index 9d60505c4..863586cd3 100644
--- a/scripts/wrap/parse.py
+++ b/scripts/wrap/parse.py
@@ -510,7 +510,7 @@ def get_args( tu, cursor, include_fz_context=False, skip_first_alt=False, verbos
     #
     if verbose:
         jlib.log( '## Looking at args of {cursor.spelling=}')
-    key = tu, cursor.location.file, cursor.location.line, include_fz_context, skip_first_alt
+    key = tu, cursor, include_fz_context, skip_first_alt
     ret = get_args_cache.get( key)
     if not verbose and state.state_.show_details(cursor.spelling):
         verbose = True
-- 
2.51.0.432.gc8867dcbdc

